portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -58,29 +58,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    BufferedReader body = request.getReader();
-    String content = body.readLine();
-    int endOfNameIndex = content.indexOf("*");
-    if (endOfNameIndex != 0 && endOfNameIndex != (content.length() -1 )) {
-      // Get name and text
-      String name = content.substring(0, endOfNameIndex++);
-      String text = content.substring(endOfNameIndex);
+    // Get name and text
+    String name = request.getParameter("name-input");
+    String text = request.getParameter("text-input");
```

Comment:
You might want to do some validation of the inputs - make sure these are both populated with non-null, non-empty strings (and maybe throw an IllegalArgumentException with a useful message if they don't have valid values) - unless you want to allow that, of course. 

Input validation is a good best practice when writing serverside code.

Reply:
fixed